oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/JwtDecoderProviderConfigurationUtils.java

```diff
@@ -90,7 +90,8 @@ static void validateIssuer(Map<String, Object> configuration, String issuer) {
 			List<? extends JWK> jwks = jwkSource.get(new JWKSelector(jwkMatcher), null);
 			for (JWK jwk : jwks) {
 				if (jwk.getAlgorithm() != null) {
-					jwsAlgorithms.add((JWSAlgorithm) jwk.getAlgorithm());
+					JWSAlgorithm jwsAlgorithm = JWSAlgorithm.parse(jwk.getAlgorithm().getName());
```

Comment:
Will you please add a test that confirms the fix? It should fail without your change and pass with your change.

Also in that test, would you please prefix it with the issue number like so:

```java
// gh-9651
@Test
public void getSignatureAlgorithmsWhenAlgorithmThenParses() {
 // ...
}
```

Reply:
added a test and updated the commit message